ground-control/internal/server/handlers.go

```diff
@@ -869,6 +869,183 @@ func (s *Server) getGroupHandler(w http.ResponseWriter, r *http.Request) {
 	WriteJSONResponse(w, http.StatusOK, result)
 }
 
+func (s *Server) deleteGroupHandler(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	groupName := vars["group"]
+
+	group, err := s.dbQueries.GetGroupByName(r.Context(), groupName)
+	if err != nil {
+		log.Println(err)
+		err := &AppError{
+			Message: "Error: Group Not Found",
+			Code:    http.StatusNotFound,
+		}
+		HandleAppError(w, err)
+		return
+	}
+
+	tx, err := s.db.BeginTx(r.Context(), nil)
+	if err != nil {
+		log.Println(err)
+		err := &AppError{
+			Message: "Error: Failed to delete group",
+			Code:    http.StatusInternalServerError,
+		}
+		HandleAppError(w, err)
+		return
+	}
+
+	q := s.dbQueries.WithTx(tx)
+
+	defer func() {
+		if p := recover(); p != nil {
+			tx.Rollback()
+			panic(p)
+		} else if err != nil {
+			tx.Rollback()
+		}
+	}()
+
+	// Remove the group from all associated satellites
+	satellites, err := s.dbQueries.GroupSatelliteList(r.Context(), group.ID)
+	if err != nil {
+		log.Println(err)
+		err := &AppError{
+			Message: "Error: Failed to list satellites for group",
+			Code:    http.StatusInternalServerError,
+		}
+		HandleAppError(w, err)
+		tx.Rollback()
+		return
+	}
+
+	for _, satellite := range satellites {
+		if err := s.dbQueries.RemoveSatelliteFromGroup(r.Context(), database.RemoveSatelliteFromGroupParams{
```

Comment:
we are not using the transaction
```suggestion
		if err := q.RemoveSatelliteFromGroup(r.Context(), database.RemoveSatelliteFromGroupParams{
```